### PR TITLE
Configure Retries for 429

### PR DESCRIPTION
**Pull request by phillc**
Eventually this should become pkg/retries and most functions become exportable.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -158,24 +158,11 @@ func (c *Client) SetToken(token string) *Client {
...
-				linodeBusy := ok && apiError.Error() == "Linode busy."
-				retry := r.StatusCode() == http.StatusBadRequest && linodeBusy
-				if retry {
-					log.Printf("[INFO] Received error %s - Retrying", apiError)
-				}
-				return retry
-			},
-		)
-
+// SetRetries adds retry conditions for Linode Busy. errors and 429s.efault to the given duration.
```
Comment (Charliekenney23) [inline client.go]: Nit: typo here

File: retries.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+type RetryConditional resty.RetryConditionFunc
+
+var retryConditionals []RetryConditional
+
+// Configures resty to
+// lock until enough time has passed to retry the request as determined by the Retry-After response header.
+// If the Retry-After header is not set, we fall back to value of SetPollDelay.
+func configureRestyRetries(resty *resty.Client) {
+	resty.
+		SetRetryCount(1000).
```
Comment (Charliekenney23) [inline retries.go]: I think to default to these loose constraints is fine for now. We should probably circle back and provide a way for consumers to customize the calls to `SetRetryCount` and `SetretryMaxWaitTime`, and merge with defaults.

File: client.go [hunk 3]
```diff
@@ -158,24 +158,11 @@ func (c *Client) SetToken(token string) *Client {
...
+func (c *Client) SetRetries() *Client {
+	addRetryConditional(linodeBusyRetryCondition)
+	addRetryConditional(tooManyRequestsRetryCondition)
```
Comment (Charliekenney23) [inline client.go]: Perhaps these 2 retry condition predicates should be added in `init`? As it stands now, these will be duplicated every time a `linodego.New()` is called.

File: retries.go (hunk 2 above)
Comment (Charliekenney23) [reply retries.go]: I just reread your description and you covered this. My bad!

File: client.go (hunk 3 above)
Comment (phillc) [reply client.go]: good catch. Maybe the array belongs on the client struct

Comment (jnschaeffer) [reply client.go]: Personally I think these would be good as options in the constructor, but we don't really have a notion of a config object that could be passed in currently.

Comment (phillc) [reply client.go]: If designed from the ground up, perhaps, but most configuration today is handled by method calls on the client

**Pull request merged**